### PR TITLE
feat(gsdk): update subxt to 0.29.0

### LINKED ISSUE #2666: Update subxt to 0.29.0
### Problem to Solve

#2667 generates runtime types with `subxt-v0.28.0` since `0.28.0` just supported generate only runtime types.

### Possible Solution

tried to do it but found it requires to modify a lot of logic, so workaround inside #2667, wait for `0.29.0` if not have to do this

--- 

EDITED: update it to `v0.29.0`

### Notes

_No response_

### PR DESCRIPTION
**Pull request by clearloop**
Resolves #2666 

- [x] fix the decoding of dispatch error
- [x] update storage queries
- [x] implement event parser manually
- [x] parse pallet name and call name from payload
- [x] trim the generated file

The size of `generated.rs` is growing during the updates of `subxt`, could be even bigger in the future releases, ( sure we can contribute to subxt to prevent this happens, but it is not 100% necessary ), note that #2744 will increase the size of `generated.rs` as well )))

so, **what changes made in this PR grows the size of the `generated.rs`?**

1. `fellowship_referenda` now declared as `Instance2` in the runtime which made this runtime metadata generating a new set of runtime types for `pallet_referenda`
2. The event decoding logic has been changed in subxt ( still can hack it but seems not necessary ), now we need to add one more trait `DecodeAsType` for the runtime types
3. the absolute path of the traits are long, there is a solution creating a new crate just do the re-export jobs, and import that crate in `gsdk` with short package name as well, but not 100% necessary as well ))

@gear-tech/dev

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -149,7 +149,22 @@ serde = "^1"
...
+# unpublished v0.29.0 from the main repo with fixes.
+#
+# 2. subxt v0.29.0 upgrades the substrate dependencies which are not
+# compatible with our current dependencies.
+#
+# 3. changing but pactching the source here for making these work out of
+# workspace.
+#
+# 4. subxt-metadata and subxt-codegen are just used by gsdk-codegen for now
+# gahtering them here for easy management.
```
Comment (shamilsan) [inline Cargo.toml:164]:
```suggestion
# gathering them here for easy management.
```

**Review by shamilsan — APPROVED** [review]
Awesome 🔥

**Comment by NikVolf** [comment]
@DennisInSky try to review please?

File: scripts/src/clippy.sh [hunk 2]
```diff
@@ -22,9 +22,14 @@ EOF
 gear_clippy() {
   EXCLUDE_PACKAGES="--exclude gear-runtime --exclude runtime-fuzzer --exclude runtime-fuzzer-fuzz"
   INCLUDE_PACKAGES="-p gear-runtime -p runtime-fuzzer -p runtime-fuzzer-fuzz"
+  ALLOW_LINTS="-A clippy::redundant-closure-call -A clippy::needless-return"
```
[the remark this answers is not in the document]
Comment (shamilsan) [reply scripts/src/clippy.sh:25]: Some of clippy warnings are related to `parity-scale-codec` v3.6.0. Please update it to v3.6.1 (here is the fixing PR: https://github.com/paritytech/parity-scale-codec/pull/455).

**Review by DennisInSky — APPROVED** [review]

**Pull request merged**